### DISCUSSION TITLE
allow IEnumerables<T> as type for "params" parameter

### DISCUSSION THREAD

**Discussion by JeReT** [dotnet/csharplang#5700] [Language Ideas] [answered]
I really like the `params` keyword and use it whenever it makes sense.
However, there are relatively often situations where I would like to pass an `IEnumerable<T>` as params parameter.
I can do `myEnumerable.ToArray()` but I have concerns if this is good for performance.
Alternatively, I can do an overload:
```csharp
void MyMethod(params char[] parameters) => MyMethod(parameters as IEnumerable<char>);
void MyMethod(IEnumerable<char> parameters) { }
```
But this is also not nice because of the required cast and in general the extra code I have to write.

I would like to have the option to write
```csharp
void MyMethod(params IEnumerable<char> parameters) { }
```
in addition to the existing
```csharp
void MyMethod(params char[] parameters) { }
```

calling the function should be similar as with the array:
```csharp
// allow nicely concatenated parameters
char a, b, c;
MyMethod(a, b, c); 

// allow an object of type IEnumerable
MyMethod("Hello".Where(o => o != 'o'));
```

**Comment by bernd5** [comment, marked as answer]
https://github.com/dotnet/csharplang/blob/main/proposals/params-span.md

**Reply by svick** [reply]
In addition to that, this is also a duplicate of https://github.com/dotnet/csharplang/issues/179.

**Comment by Richiban** [comment]
@JeReT I know it's not 100% what you're asking for (I would also very much like `params IEnumerable`) but it will get better if and when we get collection literals.

That will allow us to at least write 

```cs
SomeMethod ([1, 2, 3]);
```

for any method that accepts an `IEnumerable`, which isn't bad; it's pretty close to the

```cs
SomeMethod(1, 2, 3);
``` 

that we'd like.
